### PR TITLE
pull_request_template

### PR DESCRIPTION
**Pull request by CordlessWool**
# Description

## Approval for review
- [ ] All points were discussed with the ticket creator, support-team or product owner. The code upholds all quality guidelines from the PR-template.

> Notice: Please remove the WIP label if the PR is ready to review, otherwise nobody will review it.

### Link to Definiton of Done
More and detailed information on the *definition of done* can be found [on Confluence](https://docs.schul-cloud.org/pages/viewpage.action?pageId=92831762)

### CODE REVIEW

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 1]
```diff
@@ -0,0 +1,63 @@
+# Description
+<!--
+  This is a template to give as much as possible Informations to the pull request, to help the person which will do the review and is a checklist for you. Points to recognize are set in the comments, please read and keep them in mind:
```
Comment (Parnswir) [inline .github/PULL_REQUEST_TEMPLATE.md:3]:
```suggestion
  This is a template to add as many information as possible to the pull request, to help reviewer and as a checklist for you. Points to remember are set in the comments, please read and keep them in mind:
```

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+  
+    - Code should be selfexplain and share your knowlege with others
```
Comment (Parnswir) [inline .github/PULL_REQUEST_TEMPLATE.md:5]:
```suggestion
    - Code should be self-explanatory and share your knowledge with others
```

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+    - Document code that is not selfexplain
```
Comment (Parnswir) [inline .github/PULL_REQUEST_TEMPLATE.md:6]:
```suggestion
    - Document code that is not self-explanatory
```

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 4]
```diff
@@ -0,0 +1,63 @@
...
+    - Think about Bugs and keep security in mind
```
Comment (Parnswir) [inline .github/PULL_REQUEST_TEMPLATE.md:7]:
```suggestion
    - Think about bugs and keep security in mind
```

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 5]
```diff
@@ -0,0 +1,63 @@
...
+    - Write tests (Unit and Integration), also for error cases
+    - Main logic should hided behind the api, never trust the client
```
Comment (Parnswir) [inline .github/PULL_REQUEST_TEMPLATE.md:9]:
```suggestion
    - Main logic should hidden behind the api, never trust the client
```

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 6]
```diff
@@ -0,0 +1,63 @@
...
+    - Visibile changes should be discussed with the UX-Team from the begining of Programming, they also have to accept them at the end
```
Comment (Parnswir) [inline .github/PULL_REQUEST_TEMPLATE.md:10]:
```suggestion
    - Visible changes should be discussed with the UX-Team from the begining of development; they also have to accept them at the end
```

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 7]
```diff
@@ -0,0 +1,63 @@
...
+    - Keep changes also in changelog
```
Comment (Parnswir) [inline .github/PULL_REQUEST_TEMPLATE.md:11]:
```suggestion
    - Keep the changelog up-to-date
```

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 8]
```diff
@@ -0,0 +1,63 @@
...
+    - Remove not needed lines or set them as comment
```
Comment (Parnswir) [inline .github/PULL_REQUEST_TEMPLATE.md:12]:
```suggestion
    - Leave the code cleaner than you found it. Remove unnecessary lines. Listen to the linter.
```

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 9]
```diff
@@ -0,0 +1,63 @@
...
+-->
+
+## Links to Tickets or other pull requests
+<!--
+Base Links to copie from
```
Comment (Parnswir) [inline .github/PULL_REQUEST_TEMPLATE.md:17]:
```suggestion
Base links to copy
```

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 10]
```diff
@@ -0,0 +1,63 @@
...
+-->
+
+## Changes
+<!--
+  Short notice if a ticket exists, more detailed if not
+-->
+
+[//]: ## Datasecurity
+<!--
+  Notice about model changes, logging of user data and other user data stuff, should be noticed here. If you are not sure if it is relevant, ask the datasecureity team.
```
Comment (Parnswir) [inline .github/PULL_REQUEST_TEMPLATE.md:29]:
```suggestion
  Notice about model changes, logging of user data and other user data stuff, should be noted here. If you are not sure if it is relevant, ask the data-security team. 
```

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 11]
```diff
@@ -0,0 +1,63 @@
...
+
+[//]: ## Datasecurity
+<!--
+  Notice about model changes, logging of user data and other user data stuff, should be noticed here. If you are not sure if it is relevant, ask the datasecureity team. 
+
+-->
+
+[//]: ## Deployment
+<!--
+  Keep in mind to change seed data, if changes are done by migration scripts.
```
Comment (Parnswir) [inline .github/PULL_REQUEST_TEMPLATE.md:35]:
```suggestion
  Keep in mind to changes to seed data, if changes are done by migration scripts.
```

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 12]
```diff
@@ -0,0 +1,63 @@
...
+  Changes to the infrastructure have to discussed with the devops
+
+  This point should includes following informations:
+  - Envirement variables like FEATURE_XY=true
+  - Migration scripts to run, to get it run
```
Comment (Parnswir) [inline .github/PULL_REQUEST_TEMPLATE.md:40]:
```suggestion
  - Migration scripts to run, other requirements
```

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 13]
```diff
@@ -0,0 +1,63 @@
...
+-->
+
+[//]: ## New Repos, NPM pakages or vendor scripts
+<!--
+  Keep in mind the stability, performance, aktivity, actuality and author.
```
Comment (Parnswir) [inline .github/PULL_REQUEST_TEMPLATE.md:45]:
```suggestion
  Keep in mind the stability, performance, activity and author.
```

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 14]
```diff
@@ -0,0 +1,63 @@
...
+
+  Discripe why it is needed.
```
Comment (Parnswir) [inline .github/PULL_REQUEST_TEMPLATE.md:47]:
```suggestion
  Describe why it is needed.
```

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 15]
```diff
@@ -0,0 +1,63 @@
...
+-->
+
+## Screenshots of changes
+<!--
+  only needed for visiual stuff
```
Comment (Parnswir) [inline .github/PULL_REQUEST_TEMPLATE.md:52]:
```suggestion
  only needed for visual changes
```

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 16]
```diff
@@ -0,0 +1,63 @@
...
+
+  If visisual changes exists, work from beginning thogheter with UI/UX
```
Comment (Parnswir) [inline .github/PULL_REQUEST_TEMPLATE.md:54]:
```suggestion
  If visual changes exist, work together with UI/UX from from beginning
```

**Comment by tofixx** [comment]
I would just recommend a answerable style that gives concrete questions that may be deleted ow answered like in https://github.com/schul-cloud/commons/pull/5
- As inspiration I like the steps in https://gist.github.com/Lordnibbler/11002759
- I would add a changelog hint
- see https://github.com/devspace/awesome-github-templates for more samples

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 17]
```diff
@@ -0,0 +1,63 @@
...
+-->
+
+## Approval for review
+[ ] All points were discussed with the ticket creator, support-team or product owner. The code resolved all quaility giddeline writen in comments.
```
Comment (Parnswir) [inline .github/PULL_REQUEST_TEMPLATE.md:58]:
```suggestion
[ ] All points were discussed with the ticket creator, support-team or product owner. The code upholds all quality guidelines from the PR-template.
```

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 18]
```diff
@@ -0,0 +1,63 @@
...
+
+> Notice: Please remove the WIP label if the PR is ready to review, otherwise nobody will review it.
+
+## Link to Definiton of Done
+More and detailed Informations to the definition of Done are[in Confluence](https://docs.schul-cloud.org/pages/viewpage.action?pageId=92831762)
```
Comment (Parnswir) [inline .github/PULL_REQUEST_TEMPLATE.md:63]:
```suggestion
More and detailed information on the *definition of done* can be found [on Confluence](https://docs.schul-cloud.org/pages/viewpage.action?pageId=92831762)
```

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 19]
```diff
@@ -0,0 +1,64 @@
...
+-->
+
+## New Repos, NPM pakages or vendor scripts
+<!--
+  Keep in mind the stability, performance, activity and author.
+
+  Describe why it is needed.
+-->
+
+## Screenshots of changes
```
Comment (dominikbruechner) [inline .github/PULL_REQUEST_TEMPLATE.md:51]:
```suggestion
## Screenshots of UI changes
```

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 20]
```diff
@@ -0,0 +1,64 @@
...
+<!--
+  only needed for visual changes
+
+  If visual changes exist, work together with UI/UX from beginning
```
Comment (dominikbruechner) [inline .github/PULL_REQUEST_TEMPLATE.md:55]:
```suggestion
  If visual changes exist, work together with UI/UX from beginning/ping UX with final PR
```

File: .github/PULL_REQUEST_TEMPLATE.md (hunk 19 above)
Comment (dominikbruechner) [reply .github/PULL_REQUEST_TEMPLATE.md:51]: Can we move this part up?

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 21]
```diff
@@ -0,0 +1,64 @@
...
+-->
+
+## New Repos, NPM pakages or vendor scripts
+<!--
+  Keep in mind the stability, performance, activity and author.
+
+  Describe why it is needed.
+-->
+
+## Screenshots of UI changes
```
Comment (dominikbruechner) [inline .github/PULL_REQUEST_TEMPLATE.md:51]: Can we move this part up?

**Pull request merged**